staging/src/k8s.io/legacy-cloud-providers/vsphere/nodemanager.go

```diff
@@ -201,7 +204,23 @@ func (nm *NodeManager) DiscoverNode(node *v1.Node) error {
 				if vm != nil {
 					klog.V(4).Infof("Found node %s as vm=%+v in vc=%s and datacenter=%s",
 						node.Name, vm, res.vc, res.datacenter.Name())
-
+					var vmconfig *types.VirtualMachineConfigInfo
```

Comment:
Can we add a unit test for this version parsing logic?

Reply:
added unit tests for this 

```
=== RUN   TestVMX15Deprecated
--- PASS: TestVMX15Deprecated (0.00s)
=== RUN   TestVMX14Deprecated
--- PASS: TestVMX14Deprecated (0.00s)
=== RUN   TestVMX13Deprecated
--- PASS: TestVMX13Deprecated (0.00s)
=== RUN   TestVMX11Deprecated
--- PASS: TestVMX11Deprecated (0.00s)
=== RUN   TestVMX17Deprecated
--- PASS: TestVMX17Deprecated (0.00s)
=== RUN   TestVMX18Deprecated
--- PASS: TestVMX18Deprecated (0.00s)
```